lbuild/environment.py

```diff
@@ -323,6 +324,20 @@ def outpath(self, *path, basepath=None):
 
         return os.path.join(self.__outpath, basepath, *path)
 
+    def projectpath(self, *path):
+        """Relocate given path to the path, where the project invocation happened"""
+        return os.path.join(self.__project_path, *path)
+
+    def projectoutpath(self, *path):
+        """ Relocate given path to the outpath in respect to where the project invocation happened"""
+        return self.outpath("", basepath="")
+    
+    def relprojectoutpath(self, *path):
+        """ Relocate given path to the outpath in respect to where the project invocation happened"""
+        relative = self.projectpath("")
+        path = os.path.join(self.projectoutpath(""), *path)
+        return os.path.relpath(os.path.abspath(path), os.path.abspath(relative))
+
```

Comment:
Hm, project path sound like it's relative to the `project.xml`, but you can call `lbuild -c dir/project.xml --path build` or call lbuild without an config file entirely.

But `env.cwdpath()` sound dumb too, but it's perhaps clearer than `env.lbuildpath()`, `env.callpath()`, `env.workdir()`?

Reply:
Lbuild invocation happens (by convention) inside the project root path. At least everything else in lbuild is done in respect to where PWD is. So PWD defines where the application root(here called project) is. But yeah, I was struggling too. Well go with whatever sounds better for you.

Reply:
The name suggests it's relative to the `project.xml` but the implementation uses `cwd`. We should keep the implementation at cwd (and later add a -C switch to lbuild to move the cwd) and rename this to `env.cwdpath()`. It's not the most pretty name, but it describes exactly what it does.

Reply:
Okay, I will rename these functions. It will probably be something like this:
 ```python
def cwdpath(self, *path):
    # return absolute path of wherever lbuild was called from
```

 ```python
def cwdoutpath(self, *path):
    # return absolute path of the directory lbuild will put its output files
```

 ```python
def relcwdoutpath(self, *path):
    # return relative path of 'path' in relation to wherever lbuild was called from 
```
   
As stands inside the other pr(modm scons), these are sadly all needed because the second module has to put files and references to absolute paths of the lbuild cwd, but also relative to the defined outdir (so that scons will simply find whatever was generate via lbuild) of lbuild